src/NetAnalyzers/CSharp/Microsoft.NetCore.Analyzers/Runtime/CSharpForwardCancellationTokenToInvocations.Analyzer.cs

```diff
@@ -0,0 +1,29 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis;
+using Microsoft.NetCore.Analyzers.Runtime;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Microsoft.NetCore.CSharp.Analyzers.Runtime
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public sealed class CSharpForwardCancellationTokenToInvocationsAnalyzer : ForwardCancellationTokenToInvocationsAnalyzer
+    {
+        protected override SyntaxNode? GetMethodNameNode(SyntaxNode invocationNode)
```

Comment:
It would be really good to have this as a language specific extension method, but we do not have any place to put it right now. I will create separate C# and VB specific utilities/extensions projects in a follow-up PR and move it there. I believe all our analyzers that flag the entire invocation expression node should switch to using this approach.